### ISSUE TITLE
Automerge.view returns the wrong data when given the initial head as an argument

### ISSUE THREAD

**Issue by rongoro** [automerge/automerge#691]
I'm upgrade my application written with "automerge":"1.0.1-preview.7" to "@automerge/automerge": "^2.0.3" and I'm working through the broken tests.  The APIs aren't the same, but I believe I found a regression. The view of a document is not correct if you use the heads of the initial document.

The code below replicates the problem. 

```
import * as Automerge from "@automerge/automerge";

interface TestValue {
  foo: Automerge.Counter;
}

const initialValue: TestValue = {
  foo: new Automerge.Counter(0),
};

// Because from() accepts a Record<string, unknown> type instead of any object
// the typing is a bit disagreeable.
const initialDoc = Automerge.from(initialValue as any);

const doc2 = Automerge.change(initialDoc, (doc) => {
  doc.foo.increment(1);
});

// make a view frome the heads of the initial document
const view = Automerge.view(doc2, Automerge.getBackend(initialDoc).getHeads());

// these should all be true
console.log(initialDoc.foo.value === 0);
console.log(doc2.foo.value === 1);
console.log(view.foo.value === 0); // but this one is returning false
```

I'll just mention, in case it helps with debugging, that my original test uses a more involved document and does many changes with multiple actors. It looked like only the Counter type in a document would show the wrong version of the data. So I thought maybe the problem is Counter specific, but in the above simple example, if I add an array it also has the same problem. The code below is a slightly adjusted version of my original test. It's really ugly, in need of a refactor, and the typing isn't quite right, but it reproduces the problem where only the Counter has the wrong data in the view.

```
import * as Automerge from "@automerge/automerge";

export const getVersion = (doc: Automerge.Doc<any>): string => {
  const backend = Automerge.getBackend(doc);

  return JSON.stringify(backend.getHeads().sort());
};

export const getCRDTAtVersion = <T>(
  doc: Automerge.Doc<T>,
  versionString: string
): Automerge.Doc<T> | null => {
  const version = JSON.parse(versionString);

  const prev = Automerge.view(doc, version);
  return prev;
};

const testString = "TestString";
const testArray = [1, 2, 3, "foo", "bar"];
// const testMap = { a: 1, b: 2, c: 3, d: "four", e: "five" };

const actors = ["a1", "a2", "a3", "a4"];

const documents: Automerge.Doc<any> = [];
const documentVersions: string[] = [];

interface TestValue {
  first: Automerge.Text;
  second: string[];
  third: {};
  fourth: Automerge.Counter;
}

const initialValue: TestValue = {
  first: new Automerge.Text(""),
  second: [],
  third: {},
  fourth: new Automerge.Counter(0),
};

const currentDoc = Automerge.from(initialValue as any, { actor: "a1" });

const actorDocs = actors
  .map((a) => {
    return { [a]: Automerge.load(Automerge.save(currentDoc), a) };
  })
  .reduce((prev, next) => {
    return { ...prev, ...next };
  });

documents.push(currentDoc);
documentVersions.push(getVersion(currentDoc));

let counter = 0;

const doChange = () => {
  let newDoc: Automerge.Doc<any>;
  if (counter % 5 === 0) {
    newDoc = actors
      .map((actor) => {
        return actorDocs[actor];
      })
      .reduce(
        (prevAutoDoc: Automerge.Doc<any>, curAutoDoc: Automerge.Doc<any>) => {
          return Automerge.merge(prevAutoDoc, curAutoDoc);
        }
      );
    actorDocs[actors[0]] = newDoc;
  } else if (counter % 4 === 0) {
    const curActor = actors[counter % actors.length];
    const nextActor = actors[(counter + 1) % actors.length];
    newDoc = Automerge.merge(actorDocs[curActor], actorDocs[nextActor]);
    actorDocs[curActor] = newDoc;
  } else if (counter % 13 === 0) {
    const curActor = actors[counter % actors.length];
    const workingDoc = actorDocs[curActor];
    newDoc = Automerge.change(workingDoc, (doc: any) => {
      doc.fourth.decrement(counter % 3);
    });
    actorDocs[curActor] = newDoc;
  } else {
    const curActor = actors[counter % actors.length];
    const workingDoc = actorDocs[curActor];
    newDoc = Automerge.change(workingDoc, (doc: any) => {
      doc.first.insertAt(0, testString[counter % (testString.length - 1)]);
      doc.second.insertAt(0, testArray[counter % (testArray.length - 1)]);
      doc.third[testString[counter % (testString.length - 1)]] =
        testArray[counter % (testArray.length - 1)];
      doc.fourth.increment(counter % 4);
    });
    actorDocs[curActor] = newDoc;
  }
  documents.push(newDoc);
  documentVersions.push(getVersion(newDoc));
  counter += 1;
};

for (let i = 0; i < 100; i++) {
  doChange();
}

const finalDoc = Object.keys(actorDocs)
  .map((actor) => {
    return actorDocs[actor];
  })
  .reduce((prev: Automerge.Doc<any>, next: Automerge.Doc<any>) => {
    return Automerge.merge(prev, next);
  });

for (let y = 0; y < documents.length; y++) {
  const workingVersion = documentVersions[y];
  const expectedDocument = documents[y];

  const testDoc = getCRDTAtVersion(finalDoc, workingVersion);

  if (
    testDoc === null ||
    testDoc.fourth.value !== expectedDocument.fourth.value
  ) {
    console.log("Failed: ", y, workingVersion, expectedDocument, testDoc);
  }
}
```

**Comment by alexjg** [comment]
I believe the issue here is the use of `Automerge.getBackend(..).getHeads()`. If I change your code to use `Automerge.getHeads`, this works as expected (the third console.log is `true`). Like so:

```typescript
import * as Automerge from "@automerge/automerge";

interface TestValue {
  foo: Automerge.Counter;
}

const initialValue: TestValue = {
  foo: new Automerge.Counter(0),
};


const initialDoc = Automerge.from(initialValue as any);

const doc2 = Automerge.change(initialDoc, (doc) => {
  doc.foo.increment(1);
});

// make a view frome the heads of the initial document
// note here we use Automerge.getHeads rather then Automerge.getBackend(initialDoc).getHeads()
const view = Automerge.view(doc2, Automerge.getHeads(initialDoc)); 

// these should all be true
console.log(initialDoc.foo.value === 0);
console.log(doc2.foo.value === 1);
console.log(view.foo.value === 0); // this now prints true
```

**Comment by alexjg** [comment]
Can you try with `2.1.0-alpha.8`?

**Comment by rongoro** [comment]
Yeah, the problem seems to be fixed in 2.1.0-alpha.8. I'll continue my migration using that version and let you know if I run into other problems.
